feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -214,56 +207,46 @@ func verifyNodeConfig(t *testing.T, node gnmi.DeviceOrOpts, port *ondatra.Port,
 	}
 }
 
-func waitFor(fn func() bool, t testing.TB) {
-	start := time.Now()
-	for {
-		done := fn()
-		if done {
-			t.Logf("Expected LLDP Stats/Metric received")
-			break
-		}
-		if time.Since(start) > 65*time.Second {
-			t.Error("Timeout while waiting for expected stats...")
-		}
-		time.Sleep(500 * time.Millisecond)
-	}
-}
-
 // checkOtgLldpMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
-func checkOtgLldpMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) bool {
-	otgutils.LogLLDPMetrics(t, otg, c)
+func checkOtgLldpMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
 	for _, lldp := range c.Lldp().Items() {
-		lldpName := lldp.Name()
-		lldpState := gnmi.Get(t, otg, gnmi.OTG().LldpInterface(lldpName).State())
-		lldpStateCounters := lldpState.GetCounters()
-
-		if lldpStateCounters.GetFrameOut() == 0 {
-			return false
+		lastValue, ok := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameOut().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
```

Comment:
I think you don't have to have time.Sleep. gnmi.Watch handles timing for you.

See this example usage: https://github.com/openconfig/featureprofiles/blob/c759e3d03a11ca92b4addccf143749c4339f5705/feature/interface/aggregate/otg_tests/balancing_test/balancing_test.go#L225